meson.build

```diff
@@ -220,6 +220,7 @@ if macos_app_bundle
     datadir = join_paths(bundle_dir, 'Contents', 'Resources')
     bindir = join_paths(bundle_dir, 'Contents', 'MacOS')
     config.set('TAISEI_BUILDCONF_RELATIVE_DATA_PATH', true)
+    config.set('TAISEI_DARWIN_SOURCE', true)
```

Comment:
This should be called `TAISEI_BUILDCONF_DARWIN_SOURCE` for consistency if we're doing it this way. It's also under the wrong conditional, because `macos_app_bundle` can be `false` if the `macos_bundle` option is disabled (you get a traditional unix-like directory structure instead of an app bundle in that case).

This should be under an `if host_machine.system() == 'darwin'` clause — that'll cover macOS, iOS, and other Apple systems in case someone decides to make a port. Alternatively, I think it's fine to just check for the `__APPLE__` macro in `util/compat.h` instead.